src/lib/Libutil/misc_utils.c

```diff
@@ -2259,3 +2260,67 @@ get_msvr_mode(void)
 {
 	return 0;
 }
+
+/**
+ * @brief
+ * 		state_char2int - return the state from character form to int form.
+ *
+ * @param[in]	stc	-	state in character form
+ *
+ * @return	state in int form
+ * @retval	-1	: failure
+ */
+
+int
+state_char2int(char stc)
+{
+	int  i;
+	char statechars[] = "TQHWREXBMF";
```

Comment:
but we don't want to return the ascii numeric of the state char, we want to return numbers between 0 and 10, corresponding to the macros JOB_STATE_\<name\>. Also, I think we'd eventually want to get rid of this, store states in db in char instead of int, then we can get rid of numeric states altogether.

Reply:
Yes, we will need to generate this array as soon as the daemon start but the search will be O(1).